### PR TITLE
Authentication Support for Jenkins Users

### PR DESCRIPTION
**Pull request by sladyn98**
This PR deals with the addition of Authentication Support in order to validate users before grabbing the schema.

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import * as vscode from 'vscode';
 
 const rp = require("request-promise");
 var fs = require("fs");
+const request = require("request");
```
Comment (timja) [inline src/extension.ts]: this isn't used

File: src/extension.ts [hunk 2]
```diff
@@ -22,8 +23,12 @@ export function activate(context: vscode.ExtensionContext) {
 		// Display a message box to the user
 		vscode.window.showInformationMessage('Welcome to the jcasc-plugin');
 		const schemaURL = vscode.workspace.getConfiguration().get('jcasc.schemaURL');
-		//Fetches the JSON Schema via a REST API Call
-		rp(schemaURL).then((result:any) => {
+		const userName = vscode.workspace.getConfiguration().get('jcasc.userName');
+		const userToken = vscode.workspace.getConfiguration().get('jcasc.userToken');
+		const url = "http://" + userName + ":" + userToken + "@" + schemaURL;
```
Comment (timja) [inline src/extension.ts]: use interpolation?
```suggestion
		const url = `http://${userName}:${userToken}@${schemaURL}`;
```

File: src/extension.ts (hunk 1 above)
Comment (sladyn98) [reply src/extension.ts]: Yeah getting rid of it :+1:

File: src/extension.ts (hunk 2 above)
Comment (sladyn98) [reply src/extension.ts]: Yeah sounds cool :)

**Review by timja — APPROVED** [review]

**Pull request merged**